nion/data/Core.py

```diff
@@ -1313,23 +1330,33 @@ def function_average_region(data_and_metadata: DataAndMetadata.DataAndMetadata,
     if not Image.is_shape_and_dtype_valid(data_shape, data_dtype) or dimensional_calibrations is None:
         return None
 
-    assert len(data_and_metadata.dimensional_shape) == 3
+    if data_and_metadata.is_sequence:
+        assert len(data_and_metadata.dimensional_shape) == 4
+    else:
+        assert len(data_and_metadata.dimensional_shape) == 3
     assert len(mask_data_and_metadata.dimensional_shape) == 2
 
     data = data_and_metadata.data
-    mask_data = mask_data_and_metadata.data
+    mask_data = mask_data_and_metadata.data.astype(numpy.bool)
```

Comment:
Why is this needed? This slows down the use of mask data. Can this be a separate patch with a test if it is triggering a problem somewhere?

Reply:
This is because the `where` keyword argument requires a boolean array. It just causes an exception if the mask is not of type `bool`.
Overall this will still be faster than the previous version which multiplied the whole multidimensional array with the mask. Now, only the rather small (2D) mask will be converted and numpy will only consider those parts of the big array for the sum that are actually needed.